### PR TITLE
BUG: use internal linkage (static variables) in move.c

### LINKED ISSUE #19706: Unable to import pandas when using the gtk3 backend in ipython
```
In [1]: %matplotlib gtk3

In [2]: import pandas
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-2-38d4b0363d82> in <module>()
----> 1 import pandas

~/.local/lib/python3.6/site-packages/pandas/__init__.py in <module>()
     56 
     57 from pandas.util._print_versions import show_versions
---> 58 from pandas.io.api import *
     59 from pandas.util._tester import test
     60 import pandas.testing

~/.local/lib/python3.6/site-packages/pandas/io/api.py in <module>()
     17 from pandas.io.stata import read_stata
     18 from pandas.io.pickle import read_pickle, to_pickle
---> 19 from pandas.io.packers import read_msgpack, to_msgpack
     20 from pandas.io.gbq import read_gbq
     21 

~/.local/lib/python3.6/site-packages/pandas/io/packers.py in <module>()
     66 
     67 from pandas.io.msgpack import Unpacker as _Unpacker, Packer as _Packer, ExtType
---> 68 from pandas.util._move import (
     69     BadMove as _BadMove,
     70     move_into_mutable_buffer as _move_into_mutable_buffer,

UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 3: invalid start byte
```

If I use tk or qt instead, this doesn't happen.

### PR DESCRIPTION
**Pull request by llllllllll**
move.c declared global variables without explicitly declaring them static, so
they had global visibility. This meant that if you linked against a shared
object that provided the same symbols, the wrong data would be read. In
particular: linking to libgtk, which provides the symbol 'methods', would cause
an import error of pandas.util._move.

- [x] closes #19706
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by llllllllll** [comment]
unclear how/if we want to test this case. Also, do you want a whatsnew for a (small) bugfix like this?

**Comment by jreback** [comment]
thanks @llllllllll this looks fine. can you add a whatsnew note, ping on green.

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -1262,6 +1262,7 @@ Categorical
 - In meth:`Series.unstack`, specifying a ``fill_value`` not present in the categories now raises a ``TypeError`` rather than ignoring the ``fill_value`` (:issue:`23284`)
 - Bug when resampling :meth:`Dataframe.resample()` and aggregating on categorical data, the categorical dtype was getting lost. (:issue:`23227`)
 - Bug in many methods of the ``.str``-accessor, which always failed on calling the ``CategoricalIndex.str`` constructor (:issue:`23555`, :issue:`23556`)
+- Bug in :mod:`pandas.utils._move` where C variables were declared with external linkage causing import errors if certain other C libraries were imported before Pandas. (:issue:`24113`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.rst]: can you move to the other section, and remove the `` :mod:`pandas.utils._move` ``

**Review by jreback — CHANGES_REQUESTED** [review]
minor point. ping on green.

File: doc/source/whatsnew/v0.24.0.rst (hunk 1 above)
Comment (llllllllll) [reply doc/source/whatsnew/v0.24.0.rst]: just remove the link, or don't mention which module the error was in?

Comment (jreback) [reply doc/source/whatsnew/v0.24.0.rst]: just remove the module, i think the statement about c libraries is ok (but move it to the 'other' section, at the bottom)

**Comment by llllllllll** [comment]
I moved the whatsnew entry to other and removed the link. Sorry about putting it in "categorical" before, I just searched for "bugs" quickly. Tests are now passing. cc @jreback

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**